sklearn/tests/test_dummy.py

```diff
@@ -543,6 +543,11 @@ def test_constant_strategy_exceptions():
                           constant=[2, 0])
     assert_raises(ValueError, clf.fit, X, y)
 
+    clf = DummyClassifier(strategy='constant', constant='not-in-dataset')
+    with pytest.raises(ValueError,
+                       match="constant='not-in-dataset'.+"
+                       "Possible values.+['class1', 'class2']"):
```

Comment:
I find this context manager difficult to read. the `+` seems to add the strings rather than part of the regex. And I would use a more clearly invalid constant name.

maybe:
```py
_MSG = "constant.*must be present in.*train.*data .* provided .* 'foo'.*Possible.*['class1', 'class2']"
with pytest.raises(ValueError, match=_MSG):
    DummyClassifer(
        strategy='constant', constant='foo',
    ).fit(X=np.empty((2,2)), y=['class1', 'class2'])
```

Reply:
Since you are modifying the test, it is a good occasion to parametrize it